### DISCUSSION TITLE
mxnet.image.imread can not correctly read jpg with orientation

### DISCUSSION THREAD

**Discussion by ksellesk** [apache/incubator-mxnet#14956] [Q&A]
See also:
https://github.com/apache/incubator-mxnet/issues/13216

This is a bug must be fixed since many jpegs has orientation.

People believe that mxnet will do right thing but waste time for debugging other thing.

[a comment by an automated account is not shown]

**Comment by ksellesk** [comment]
![IMG_2058](https://user-images.githubusercontent.com/9996886/57754105-47df8300-7720-11e9-94f0-7d9f1ec8766d.JPG)

Such as this pic

**Comment by vdantu** [comment]
@mxnet-label-bot add [question, bug]

**Comment by anirudhacharya** [comment]
@ksellesk this is the doc for opencv's imread - https://docs.opencv.org/3.0-beta/modules/imgcodecs/doc/reading_and_writing_images.html?highlight=imread#cv2.imread

can you please provide a code example or usage of reading jpg with orientation. The method signatures of mxnet's imread and opencv's imread seem pretty similar.

**Comment by ksellesk** [comment]
see  this one 

https://docs.opencv.org/3.4.6/d4/da8/group__imgcodecs.html



https://github.com/apache/incubator-mxnet/blob/master/src/io/image_io.cc#L159


why IGNORE ORIENTATION?

**Comment by anirudhacharya** [comment]
```python
b = mx.img.imread("img.jpg", flag=cv2.IMREAD_GRAYSCALE)
```
in the above code snippet except for `cv2.IMREAD_GRAYSCALE` and `cv2.IMREAD_COLOR` all the flags mentioned below errors out in mxnet while they are supported in opencv.

```bash
cv2.IMREAD_UNCHANGED
cv2.IMREAD_GRAYSCALE
cv2.IMREAD_COLOR
cv2.IMREAD_ANYDEPTH
cv2.IMREAD_ANYCOLOR
cv2.IMREAD_LOAD_GDAL
cv2.IMREAD_REDUCED_GRAYSCALE_2
cv2.IMREAD_REDUCED_COLOR_2
cv2.IMREAD_REDUCED_GRAYSCALE_4
cv2.IMREAD_REDUCED_COLOR_4
cv2.IMREAD_REDUCED_GRAYSCALE_8
cv2.IMREAD_REDUCED_COLOR_8
cv2.IMREAD_IGNORE_ORIENTATION 
```

**Comment by ksellesk** [comment]
>>> img = cv2.imread("test.jpg")
>>> img.shape
(4032, 3024, 3)
>>> img = mx.image.imread("test.jpg")
>>> img.shape
(3024, 4032, 3)


you can test this pic:
https://user-images.githubusercontent.com/9996886/57754105-47df8300-7720-11e9-94f0-7d9f1ec8766d.JPG

**Comment by anirudhacharya** [comment]
here is a weird discrepancy I am facing when using opencv with python and with c++ - 
```python
cv2.imread("img.jpg", flags=cv2.IMREAD_REDUCED_GRAYSCALE_2)
```
returns an image of shape `(112, 112)`
but when I use `imread` in c++ with
```cpp
cv::imdecode(buf, cv::IMREAD_REDUCED_GRAYSCALE_2)
```
returns a matrix of shape (224,224,1). 

The original image shape is (224,224,3)

**Comment by VinceJnz** [comment]
I'm having the same problem
mx.image.imread crashes when reading portrate oriented images.
I am using mxnet 1.7.0

Has there been any fix for this?
